### PR TITLE
Allow alternate core implementation of CMAC

### PR DESCRIPTION
**Pull request by stevew817**
Allow hardware platforms with an integrated crypto accelerator capable of CMAC to do CMAC natively, instead of having to go through the mbedTLS AES/(3)DES primitive block-by-block.

### CODE REVIEW

File: library/cmac.c [hunk 1]
```diff
@@ -468,6 +470,8 @@ int mbedtls_aes_cmac_prf_128( const unsigned char *key, size_t key_length,
 }
 #endif /* MBEDTLS_AES_C */
 
+#endif /* !MBEDTLS_CMAC_ALT */
```
Comment (RonEld) [inline library/cmac.c]: Note that `cmac_generate_subkeys()` is also used in  the [self_test](https://github.com/ARMmbed/mbedtls/pull/866/files#diff-97f39c1c46fcba3c19208c069f04a40fR809). You should consider taking this function out of `MBEDTLS_CMAC_ALT` check.
I would consider putting this function within:
~~~
#if ! defined(MBEDTLS_CMAC_ALT) || defined(MBEDTLS_SELF_TEST)
~~~

**Comment by sbutcher-arm** [comment]
@RonEld - I'd like to get this merged in the next couple of weeks. Could you take @stevew817's PR, and apply your proposed fix, so we can get this going? Thanks.

**Comment by RonEld** [comment]
@sbutcher-arm  I applied my proposed fix.

**Comment by gilles-peskine-arm** [comment]
CI status: there were failures but these are known problems (timing self-test and `compat.sh` slow start) that are unrelated to this PR. So CI ok for merging purposes.

**Comment by sbutcher-arm** [comment]
Approved for merge.

@gilles-peskine-arm - Can you please correct the ChangeLog to attribute this to 'Steve Cooreman, Silicon Labs' instead of @stevew817?

Thanks.

**Pull request merged**